src/components/Search.tsx

```diff
@@ -9,35 +9,87 @@ import * as SearchUtils from '@libs/SearchUtils';
 import Navigation from '@navigation/Navigation';
 import EmptySearchView from '@pages/Search/EmptySearchView';
 import useCustomBackHandler from '@pages/Search/useCustomBackHandler';
+import CONST from '@src/CONST';
+import type {SearchColumnType} from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
+import type {SearchQuery} from '@src/types/onyx/SearchResults';
 import {isEmptyObject} from '@src/types/utils/EmptyObject';
 import isLoadingOnyxValue from '@src/types/utils/isLoadingOnyxValue';
 import SelectionList from './SelectionList';
 import SearchTableHeader from './SelectionList/SearchTableHeader';
+import type {TransactionListItemType} from './SelectionList/types';
 import TableListItemSkeleton from './Skeletons/TableListItemSkeleton';
 
+const columnNamesToPropertyMap = {
+    [CONST.SEARCH_TABLE_COLUMNS.TO]: 'to',
+    [CONST.SEARCH_TABLE_COLUMNS.FROM]: 'from',
+    [CONST.SEARCH_TABLE_COLUMNS.DATE]: 'created',
+    [CONST.SEARCH_TABLE_COLUMNS.TAG]: '',
+    [CONST.SEARCH_TABLE_COLUMNS.MERCHANT]: 'merchant',
+    [CONST.SEARCH_TABLE_COLUMNS.TOTAL]: 'amount',
+    [CONST.SEARCH_TABLE_COLUMNS.CATEGORY]: 'category',
+    [CONST.SEARCH_TABLE_COLUMNS.TYPE]: null,
+    [CONST.SEARCH_TABLE_COLUMNS.ACTION]: null,
+    [CONST.SEARCH_TABLE_COLUMNS.DESCRIPTION]: null,
+    [CONST.SEARCH_TABLE_COLUMNS.TAX_AMOUNT]: null,
+};
+
+function getSortedData(data: TransactionListItemType[], sortBy?: SearchColumnType, sortOrder?: 'asc' | 'desc') {
+    if (!sortBy || !sortOrder) {
+        return data;
+    }
+
+    const sortingProp = columnNamesToPropertyMap[sortBy] as keyof TransactionListItemType;
+
+    if (!sortingProp) {
+        return data;
+    }
+
+    // Todo sorting needs more work
+    return data.sort((a, b) => {
+        const aValue = a[sortingProp];
+        const bValue = b[sortingProp];
+
+        if (!aValue || !bValue) {
+            return 0;
+        }
+
+        if (typeof aValue === 'string' && typeof bValue === 'string') {
+            return sortOrder === 'asc' ? aValue.toLowerCase().localeCompare(bValue) : bValue.toLowerCase().localeCompare(aValue);
+        }
+
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-expect-error
+        return sortOrder === 'asc' ? aValue - bValue : bValue - aValue;
+    });
+}
+
 type SearchProps = {
-    query: string;
+    query: SearchQuery;
     policyIDs?: string;
+    sortBy?: SearchColumnType;
+    sortOrder?: 'asc' | 'desc';
```

Comment:
Can we create a separate type for `asc | desc` and reuse it in other places where we define `sortOrder`?